### PR TITLE
have Event#toMap return copy

### PR DESCRIPTION
**Pull request by talevy**

### CODE REVIEW

**Comment by colinsurprenant** [comment]
I think we need to verify current usages of `toMap()` to make sure there's no double-cloning involved like https://github.com/talevy/logstash/blob/1339b096b18de958eafb6cef08627f6a5a5c64c7/logstash-core-event-java/src/main/java/com/logstash/ext/JrubyEventExtLibrary.java#L248

**Comment by talevy** [comment]
ah, I forgot about this one that I changed in the previous PR. will update

**Comment by talevy** [comment]
@colinsurprenant thanks, updated. I see no other usages

File: logstash-core-event-java/src/test/java/com/logstash/EventTest.java [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 import java.io.IOException;
 import java.util.*;
+
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.core.IsSame.sameInstance;
```
Comment (colinsurprenant) [inline logstash-core-event-java/src/test/java/com/logstash/EventTest.java]: are we introducing a new `org.hamcrest` dependency?

Comment (talevy) [inline logstash-core-event-java/src/test/java/com/logstash/EventTest.java]: looks like it has already been included. I just autocompleted. unless I am mistaken?

Comment (talevy) [inline logstash-core-event-java/src/test/java/com/logstash/EventTest.java]: I liked the `sameInstance` assertion from there, so I started typing it. and it worked. so I stuck with it

**Comment by talevy** [comment]
Looks like it is somehow shadowed inside the `json-unit` testCompile dependency?

**Comment by colinsurprenant** [comment]
I don't really mind if we introduce a new testCompile dep but I am not sure how this can work if it is not included in `build.gradle`?

**Comment by talevy** [comment]
It is being brought in by `json-unit`

**Comment by talevy** [comment]
well. it doesn't matter anymore. I removed the code dependency

**Comment by colinsurprenant** [comment]
LGTM

**Pull request closed without merging**